File: Arrays/Find_Missing_No/optimal_sol.cpp
#include <iostream>
#include <vector>
using namespace std;

int missingNum(vector<int> &arr){
    int n = arr.size()+1;

    int sum =0;
    for(int i=0; i<n-1; i++){
        sum+=arr[i];
    }

    long long expSum = (n*1LL*(n+1))/2;

    return expSum-sum;

}
int main() {
    vector<int> arr = {8, 2, 4, 5, 3, 7, 1};  
    cout << missingNum(arr);  
    return 0;
}

File: Arrays/Find_Missing_No/optimal_sol.py
def missingNum(arr):
    n = len(arr)+1

    
    for i in range(n-1):
        totalSum = sum(arr)

    expSum = n*(n+1)//2

    return expSum - totalSum

if __name__ == '__main__':
    arr = [8, 2, 4, 5, 3, 7, 1]
    print(missingNum(arr))